packages/fxa-auth-server/lib/routes/password.js

```diff
@@ -230,54 +242,53 @@ module.exports = function (
-              return db.resetAccount(passwordChangeToken, {
-                verifyHash: verifyHash,
-                authSalt: authSalt,
-                wrapWrapKb: wrapWrapKb,
-                verifierVersion: password.version,
-                keysHaveChanged: false,
-              });
-            })
-            .then((result) => {
-              return request
-                .emitMetricsEvent('account.changedPassword', {
-                  uid: passwordChangeToken.uid,
-                })
-                .then(() => {
-                  // NOTE: Not quite sure of the difference between these two events:
-                  recordSecurityEvent('account.password_reset_success', {
-                    db,
-                    request,
-                    account: passwordChangeToken,
-                  });
-                  recordSecurityEvent('account.password_changed', {
-                    db,
-                    request,
-                    account: passwordChangeToken,
-                  });
-                })
-                .then(() => {
-                  return result;
-                });
-            });
+        async function changePassword() {
+          const authSalt = await random.hex(32);
+          const password = new Password(authPW, authSalt, verifierVersion);
+          const verifyHash = await password.verifyHash();
+          const wrapWrapKb = await password.wrap(wrapKb);
+
+          // For the time being we store both passwords in the DB. authPW is created
+          // with the old quickStretch and authPWVersion2 is created with improved 'quick' stretch.
+          let password2 = undefined;
+          let verifyHashVersion2 = undefined;
+          let wrapWrapKbVersion2 = undefined;
+          if (authPWVersion2) {
+            password2 = new Password(authPWVersion2, authSalt, verifierVersion, 2);
+            verifyHashVersion2 = await password2.verifyHash();
+            wrapWrapKbVersion2 = await password2.wrap(wrapKbVersion2);
+          }
+
+          await db.deletePasswordChangeToken(passwordChangeToken);
+
+          const result = await db.resetAccount(passwordChangeToken, {
+            authSalt: authSalt,
+            clientSalt: clientSalt,
+            verifierVersion: password.version,
+            verifyHash: verifyHash,
+            verifyHashVersion2: verifyHashVersion2,
+            wrapWrapKb: wrapWrapKb,
+            wrapWrapKbVersion2: wrapWrapKbVersion2,
+            keysHaveChanged: false,
+          });
+
+          await request.emitMetricsEvent('account.changedPassword', {
+            uid: passwordChangeToken.uid,
+          });
+
+          await recordSecurityEvent('account.password_reset_success', {
+            db,
+            request,
+            account: passwordChangeToken,
+          });
+
+          await recordSecurityEvent('account.password_changed', {
+            db,
+            request,
+            account: passwordChangeToken,
+          });
```

Comment:
Doesn't matter much but one small difference is that the `recordSecurityEvent` calls were not awaited previously.

Reply:
Ah good call out. It seems like awaiting this maybe better, unless we suspect they are slow. What do you think? Happy to go either way.